lib/spack/llnl/util/filesystem.py

```diff
@@ -337,41 +337,55 @@ def unset_executable_mode(path):
 
 
 def copy(src, dest, _permissions=False):
-    """Copies the file *src* to the file or directory *dest*.
+    """Copy the file(s) *src* to the file or directory *dest*.
 
     If *dest* specifies a directory, the file will be copied into *dest*
     using the base filename from *src*.
 
+    *src* may contain glob characters.
+
     Parameters:
-        src (str): the file to copy
+        src (str): the file(s) to copy
         dest (str): the destination file or directory
         _permissions (bool): for internal use only
+
+    Raises:
+        IOError: if src does not match any files or directories
     """
     if _permissions:
         tty.debug('Installing {0} to {1}'.format(src, dest))
     else:
         tty.debug('Copying {0} to {1}'.format(src, dest))
 
-    # Expand dest to its eventual full path if it is a directory.
-    if os.path.isdir(dest):
-        dest = join_path(dest, os.path.basename(src))
+    files = glob.glob(src)
+    if not files:
+        raise IOError("No such file or directory: '{0}'".format(src))
 
-    shutil.copy(src, dest)
+    for src in files:
+        # Expand dest to its eventual full path if it is a directory.
+        dst = dest
+        if os.path.isdir(dest):
+            dst = join_path(dest, os.path.basename(src))
 
-    if _permissions:
-        set_install_permissions(dest)
-        copy_mode(src, dest)
+        shutil.copy(src, dst)
+
+        if _permissions:
+            set_install_permissions(dst)
+            copy_mode(src, dst)
```

Comment:
What about when `len(files) == 1`? There are a lot of times when you're looking for a single file but it might contain a version or arch in the path. I think I'll set it to error when `len(files) > 1` and `dst` is not a directory.

Reply:
> There are a lot of times when you're looking for a single file but it might contain a version or arch in the path.

I'm not really sure about this case. Being defensive I'd go for erroring on a glob or being explicit in the API that only a single file is expected to match, but I don't know how frequent this case would be in practice.

Reply:
I think I made the API docs explicit. `src` can contain a glob, and `dest` can be either a file or directory, but it will raise an error if `src` matches multiple files and `dest` is a file.